### ISSUE TITLE
Add flow chart with relation between data types

### ISSUE THREAD

**Issue by stefsmeets** [hpgem/nanomesh#241]
Add a flowchart to the documentation that describes how classes are related.

GenericImage -> (Plane, Volume)
AbstractMesher -> (Mesher2D, Mesher3D)
GenericMesh -> (LineMesh, TriangleMesh, TetraMesh)

Also make sure to add the different functions that are supposed to be used in sequence

**Comment by stefsmeets** [comment]
```mermaid
  graph TD;
      A[Image data] -- 2D --> B[Plane]
      B --> C[Mesher2D]
      C --> D[MeshContainer]
      D -- 2D/3D --> E[LineMesh]
      D -- 2D/3D --> F[TriangleMesh]
      D --    3D --> G[TetraMesh]

      A[Image data] -- 2D --> K[Volume]
      K --> L[Mesher3D]
      L --> D[MeshContainer]
```

**Comment by stefsmeets** [comment]
Closed by #254
